### PR TITLE
On startup, retry connecting to zookeeper till successful or timeout

### PR DESCRIPTION
**Pull request by endika-ravn**

### CODE REVIEW

File: solr/solrj/src/java/org/apache/solr/common/util/RetryUtil.java [hunk 1]
```diff
@@ -58,6 +59,23 @@ public static void retryOnThrowable(Set<Class> classes, long timeoutms, long int
       break;
     }
   }
+
+  public static <T> T retry(Set<Class> retryOnExceptions, long timeoutms, long intervalms, Callable<T> callable) throws Exception {
```
Comment (craigmamakin) [inline solr/solrj/src/java/org/apache/solr/common/util/RetryUtil.java:76]: no javadoc?

File: solr/solrj/src/test/org/apache/solr/common/util/TestRetryUtil.java [hunk 2]
```diff
@@ -68,4 +69,64 @@ public void testRetryOnThrowable() throws Throwable {
     assertTrue(executes3.get() > 1);
   }
 
-}
+
+
+  public void testRetry() throws Throwable {
```
Comment (craigmamakin) [inline solr/solrj/src/test/org/apache/solr/common/util/TestRetryUtil.java:74]: there should be one test method per test case imo

Comment (endika-ravn) [reply solr/solrj/src/test/org/apache/solr/common/util/TestRetryUtil.java:74]: in mine too, but that's how they are in the solr project. This is a copy paste of another test with a few modifications, in the same class

File: solr/solrj/src/java/org/apache/solr/common/util/RetryUtil.java (hunk 1 above)
Comment (endika-ravn) [reply solr/solrj/src/java/org/apache/solr/common/util/RetryUtil.java:76]: done

**Pull request merged**